### ISSUE TITLE
Provide full stack trace on custom command failure

### ISSUE THREAD

**Issue by St1fle** [cypress-io/cypress#15581]
### What would you like?
I would like to see the full stack trace when a custom command fails.
At the moment I see this:
![image](https://user-images.githubusercontent.com/5892508/111780544-721ba200-88af-11eb-992b-e344a71904b5.png)

So I can see the line in the custom command that failed and I can see which command was invoked that failed, but I cannot see which line in a test was the command called from.

I'm pretty sure we already get a full stack trace for Cypress's built-in commands, taking you all the way back to the line of the test that caused them. For example if `cy.click()` fails to click an element, you get a larger stack trace including the line in the test that called the click method that failed, so I wish we would get the same just with custom commands.

### Why is this needed?
It would improve the usability of custom commands since it's time consuming to find the line on the test calling the command that failed, especially if the same test invokes the same custom command multiple times.

An example would be, we wrote a custom command that interacts with our modals and automatically selects an item from a modal with a specific data attribute tag. Because we have a lot of modals this command gets invoked multiple times per test, making it hard to figure out which exact command in the test caused the error without having to look at the logs backwards, especially if we are trying to debug an error from a CI.

**Comment by jennifer-shehane** [comment]
Related error improvements: https://github.com/cypress-io/cypress/issues/7202

**Comment by xxleyi** [comment]
Still has any plan on this enhancement?

**Comment by TheRealSamChaney** [comment]
Still an issue 4 years later. The stack trace from Cypress is useless if there are custom commands used in the test. There is no way to conveniently get from the Cypress test runner to the file and line where the failing command was actually called. Cypress just links me to the definition of the command in commands.js for example, which is not useful

**Comment by mihhail-m** [comment]
bump

**Comment by archibald1418** [comment]
Yeah so basically without a normal custom command stack trace, the only option is to copy-paste cypress query/command sequences into cases (cumbersome if the logic is big). Putting cypress command logic into a JS function doesn't work either, it just shows the calls inside the function. This is especially painful to debug in CI

**Comment by Sarcolemna** [comment]
Lack of custom command tracing has been bothering me for a while. This solution worked for me: https://gist.github.com/Sarcolemna/5669f8bd76978fb2e80cf32e2fbbe8b3

Note this was completely vibe coded with several rounds of instruction/guidance followed by testing and is beyond my engineering abilities. There may be better approaches. Please inspect and optimize if you decide to give it a try. I can't assist with debugging,

**Summary and result**
This file appends an additional error line within the thrown error Cypress returns normally. You'll get the same error showing the failure within the command file and also the line number within the spec file where a failing Cypress custom command was invoked. To illustrate:

Here we have line 197 using the custom command 'getbySel' which is the custom getter for QA tags in the project. This line from the spec file will generate a failure:

<img width="692" height="40" alt="Image" src="https://github.com/user-attachments/assets/1eb5442a-6e70-4f9b-bee3-d3d5c93b0c62" />


During a headed run this will show the following error.  It adds the "Invoked at: cypress/e2e/----/cart.cy.js:197 to the existing error message. The addition points to the corresponding failing line in the spec that used the custom command getBySel.

<img width="420" height="435" alt="Image" src="https://github.com/user-attachments/assets/f6147687-c55a-4792-a1f0-5267680ce07b" />

It works similarly in headless runs with default verbosity. In CLI it will appear in the log as:

<img width="1005" height="117" alt="Image" src="https://github.com/user-attachments/assets/fc8c6813-0a52-4a20-80da-f3886bbebad4" />

When the failure source is not from a custom command, the error message will remain the same as it was before and not include the custom command trace.

**AI made functional summary:**
1. Extends Cypress.Commands.add() to record which command names are custom (a Set lookup)
2. Listens to command:start — records each custom command's userInvocationStack (the raw JS stack trace captured at enqueue time, which contains the caller's location)
3. Listens to fail — searches the history backwards for the most recent custom command called from a .cy.js file. Extracts the webpack bundle URL and line number from userInvocationStack
4. Resolves the bundle line to the actual source line using a built-in VLQ source map decoder (fetches the bundle's source map via sync XHR on first failure, caches the mapping)
5. Appends Invoked at: <file>:<line> to err.message. Does not replace the original code frame

**If you would like to try it:**
1. Place [customCommandTracer.js](https://gist.github.com/Sarcolemna/5669f8bd76978fb2e80cf32e2fbbe8b3) in the /cypress folder within the project
2. Add to the first line of imports within your e2e.js at the top

Done and ready to run.

**Consider also:**
-Relies on userInvocationStack on command objects and err.codeFrame on errors
-Won't work for commands registered via override(). Only works on registrations from add()
-Source map fetch depends on XMLHttpRequest.  Something to maybe swap out or improve later.
